### PR TITLE
JSUI-3033 Improved accessibility for `UserFeedbackBanner` and `HeightLimiter`

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/JSUI-3033

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/SmartSnippet/UserFeedbackBanner.ts [hunk 1]
```diff
@@ -148,7 +163,9 @@ export class UserFeedbackBanner {
     }
     this.isUseful = isUseful ? UsefulState.Yes : UsefulState.No;
     this.yesButton.classList.toggle(BUTTON_ACTIVE_CLASSNAME, isUseful);
+    this.yesButton.setAttribute('aria-pressed', `${isUseful}`);
```
Comment (olamothe) [inline src/ui/SmartSnippet/UserFeedbackBanner.ts:166]: What does the `aria-pressed` do exactly ? 

Does it signify to screen reader "here's what going to happen when you click this button" ?

File: src/ui/SmartSnippet/UserFeedbackBanner.ts [hunk 2]
```diff
@@ -78,7 +84,7 @@ export class UserFeedbackBanner {
   }
 
   private buildLabel() {
-    return $$('span', { className: LABEL_CLASSNAME }, l('UsefulnessFeedbackRequest')).el;
+    return $$('span', { className: LABEL_CLASSNAME, id: this.labelId }, l('UsefulnessFeedbackRequest')).el;
```
Comment (olamothe) [inline src/ui/SmartSnippet/UserFeedbackBanner.ts:87]: Could we use a better semantic tag than `span` ?

File: src/ui/SmartSnippet/UserFeedbackBanner.ts (hunk 1 above)
Comment (btaillon) [reply src/ui/SmartSnippet/UserFeedbackBanner.ts:166]: When the `aria-pressed` attribute exists, whether `true` or `false`, it signifies the element is a toggleable button.

File: src/ui/SmartSnippet/UserFeedbackBanner.ts (hunk 2 above)
Comment (btaillon) [reply src/ui/SmartSnippet/UserFeedbackBanner.ts:87]: On its own, the semantic choice would probably be a `label`. However, a `label` cannot be associated with multiple controls. I considered using a `fieldset` and a `legend` back when I considered making those buttons `input[type="radio"]`, but I let go of the idea at the same time that I chose to make them toggleable buttons.

I'll keep them as toggleable buttons but I'll make this a `legend` and introduce a `fieldset`.

**Pull request merged**